app/models/event.rb

```diff
@@ -69,4 +71,15 @@ def time_zone_valid?
     time_zone.blank? || (ActiveSupport::TimeZone.all.map{ |t| t.tzinfo.name }.include? time_zone)
   end
 
+  def set_timezone
+    if time_zone.present? && time_zone_valid?
+      if start_date.present? && (start_date_changed? || time_zone_changed?)
+        self.start_date = start_date.to_s(:db).in_time_zone(time_zone)
+      end
+      if end_date.present? && (end_date_changed? || time_zone_changed?)
+        self.end_date = end_date.to_s(:db).in_time_zone(time_zone)
+      end
+    end
+  end
+
```

Comment:
Could we have some tests for this behaviour? :)

1. Check the times are shifted according to the time zone
2. Check the times do not shift on subsequent saves
3. Check the times DO shift if the timezone is changed